gutils/fsys.c

```diff
@@ -1159,3 +1159,37 @@ char *GFileDirName(const char *path) {
     return GFileDirNameEx(path, 0);
 }
 
+char* GFileParent(char* path) {
+    GFile *gf_path = g_file_new_for_path(path);
+    GFile *gf_path_parent = g_file_get_parent(gf_path);
+
+    char* ret = g_file_get_path(gf_path_parent);
+
+    g_object_unref(gf_path);
+    g_object_unref(gf_path_parent);
+
+    return ret;
+}
+
+// If we have two files:
+// * /tmp/ex/test.sfd (file1)
+// * /tmp/ex/images/1.jpg (to_relative)
+//
+// The return is images/1.jpg
+char* GFileRelativize(char* file1, char* to_relative) {
+   GFile *gf_file1 = g_file_new_for_path(file1);
+   GFile *gf_file1_parent = g_file_get_parent(gf_file1);
+   GFile *gf_to_relative = g_file_new_for_path(to_relative);
+
+   char* ex = g_file_get_relative_path(gf_file1_parent, gf_to_relative);
+
+   g_object_unref(gf_file1);
```

Comment:
If you absolutely must use this gio GFile stuff, then you'll need to change this to include `gobject` in the COMPONENTS list:

https://github.com/fontforge/fontforge/blob/decf26fd43251502b66c7264dc41fec3d9937289/CMakeLists.txt#L81

Reply:
Will you accept that our should I work harder to avoid needing Gio?

Reply:
Decided to require gobject for now, if that's okay feel lmk but I'm going to mark this resolved for now.